app/utils/misc.js

```diff
@@ -51,6 +51,22 @@ export function getObjArrayKey(obj, key, defaultValue = '') {
 }
 
 /**
+ * get value from key or default if not set or not a string,
+ * also works for arrays
+ *
+ * @param {Object} obj Basic object or array
+ * @param {String} key Key to look for
+ * @param {*} defaultValue Default to return if key is not set
+ */
+export function getObjArrayKeyStringOnly(obj, key, defaultValue = '') {
```

Comment:
`getObjArrayKey` is still in use elsewhere to my knowledge so I kept it.

Didn't go with the generic implementation for now; kept it as is. Might have been my coding, but the generic implementation you suggested seemed to be causing issues across the board with the metadata fields always validating incorrectly. Can revisit this if we find a need for a generic at some point in the future.